specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/AssistedQueries.json

```diff
@@ -0,0 +1,612 @@
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "QueryList": {
+      "description": "List all the queries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of queries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of queries.",
+          "items": {
+            "$ref": "#/definitions/Query"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Query": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents a Query in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/QueryProperties",
+          "description": "Query properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "QueryProperties": {
+      "description": "Describes Query properties",
+      "properties": {
+        "queryId": {
+          "description": "The id (a Guid) of the query",
+          "type": "string"
+        },
+        "correlationId": {
+          "description": "The id (a Guid) of the correlationId",
+          "type": "string"
+        },
+        "naturalLanguage": {
+          "description": "The natural language",
+          "type": "string"
+        },
+        "kustoQueryLanguage": {
+          "description": "the kusto query for a natural language",
+          "type": "string"
+        },
+        "documentType": {
+          "description": "A type of the document (is 'assistedQuery')",
+          "type": "string"
+        },
+        "executionResult": {
+          "description": "The result of the execution to get kusto query from  natural language input",
+          "type": "string"
+        }
+      },
+      "required": [
+        "naturalLanguage"
+      ],
+      "type": "object"
+    },
+    "EditedQueryList": {
+      "description": "List all the Edited Queries.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of edited queries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of edited queries.",
+          "items": {
+            "$ref": "#/definitions/EditedQuery"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object",
+      "required": [
+        "value"
+      ]
+    },
+    "EditedQuery": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
```

Comment:
>ResourceWithEtag [](http://example.com/codeflow?start=63&length=16)

same here for the resource type

Reply:
same as above. followed the other APIs as a sample. not sure if that should only apply to my new API